scripts/install

```diff
@@ -25,323 +23,269 @@ ynh_abort_if_errors
-###
-### --test_status "some command"  a custom command to check the status of the service
-###                               (only relevant if 'systemctl status' doesn't do a good job)
-###
-### --test_conf "some command"    some command similar to "nginx -t" that validates the conf of the service
-###
-### Re-calling 'yunohost service add' during the upgrade script is the right way
-### to proceed if you later realize that you need to enable some flags that
-### weren't enabled on old installs (be careful it'll override the existing
-### service though so you should re-provide all relevant flags when doing so)
+yunohost service add $app-backend --description="Backend for librephotos" --log="/var/log/$app/$app-backend.log"
+yunohost service add $app-frontend --description="Frontend for librephotos" --log="/var/log/$app/$app-frontend.log"
+yunohost service add $app-image-similarity --description="Image similarity server for librephotos" --log="/var/log/$app/$app-image-similarity.log"
+yunohost service add $app-frontend --description="Worker for librephotos" --log="/var/log/$app/$app-worker.log"
 
 #=================================================
-# START SYSTEMD SERVICE
+# START SYSTEMD SERVICES
 #=================================================
-ynh_script_progression --message="Starting a systemd service..." --time --weight=1
+ynh_script_progression --message="Starting systemd services..." --weight=1
 
-### `ynh_systemd_action` is used to start a systemd service for an app.
-### Only needed if you have configure a systemd service
-### If you're not using these lines:
-###		- Remove the section "STOP SYSTEMD SERVICE" and "START SYSTEMD SERVICE" in the backup script
-###		- As well as the section "START SYSTEMD SERVICE" in the restore script
-###		- As well as the section"STOP SYSTEMD SERVICE" and "START SYSTEMD SERVICE" in the upgrade script
-###		- And the section "STOP SYSTEMD SERVICE" and "START SYSTEMD SERVICE" in the change_url script
-
-# Start a systemd service
-ynh_systemd_action --service_name=$app --action="start" --log_path="/var/log/$app/$app.log"
-
-#=================================================
-# SETUP FAIL2BAN
-#=================================================
-ynh_script_progression --message="Configuring Fail2Ban..." --time --weight=1
-
-# Create a dedicated Fail2Ban config
-ynh_add_fail2ban_config --logpath="/var/log/nginx/${domain}-error.log" --failregex="Regex to match into the log for a failed login"
+# Start systemd services
+ynh_systemd_action --service_name=$app-backend --action="start" --log_path="/var/log/$app/$app-backend.log"
+ynh_systemd_action --service_name=$app-frontend --action="start" --log_path="/var/log/$app/$app-frontend.log"
+ynh_systemd_action --service_name=$app-image-similarity --action="start" --log_path="/var/log/$app/$app-image-similarity.log"
+ynh_systemd_action --service_name=$app-frontend --action="start" --log_path="/var/log/$app/$app-worker.log"
```

Comment:
You may be getting `502` errors due to the services not being fully started when the CI checks if the page loads. You should add a `--line_match` to the helper. Check `ynh_systemd_action` over [there](https://yunohost.org/fr/packaging_apps_helpers) to see how to add it (you need to tweak the `ynh_clean_setup`, notably).

Reply:
Thanks for the information!